InternalDocs/README.md

```diff
@@ -40,3 +40,8 @@ Program Execution
 - [Garbage Collector Design](garbage_collector.md)
 
 - [Exception Handling](exception_handling.md)
+
+Other
+---
+
+- [String Interning](string_interning.md)
```

Comment:
I think this fits in Runtime Objects. Even better if we could have a page about string implementation (and this as one of the sections in that page).